cabal-testsuite/src/Test/Cabal/Monad.hs

```diff
@@ -176,23 +178,32 @@ testArgParser = TestArgs
     <*> argument str ( metavar "FILE")
     <*> commonArgParser
 
-skip :: String -> TestM ()
-skip reason = liftIO $ do
+skipIO :: String -> IO ()
```

Comment:
For the reviewer: This does not need to be in `TestM` always, there are things we already know that we can discard a test right-away instead of having to wait for the whole test environment to set up, like the OS being windows.